### ISSUE TITLE
Internet drop crashes game

### ISSUE THREAD

**Issue by CaptainMorgan12** [Heroic-Games-Launcher/HeroicGamesLauncher#3445]
### Problem description

There is a setting to run the game offline. I believe this should be a default setting. Alternatively when there is no internet connection games downloaded should still be able to run. Why is an internet connection needed? I am not using the EOS overlay? If checking for updates is selected and you are offline and starting a game, it should not check for updates.

### Feature description

Do not crash game when connection goes from online to offline during gameplay for a game that does not have the selection made Play Offline. A banner or notification should be sufficient.

**Comment by CommandMC** [comment]
> There is a setting to run the game offline. I believe this should be a default setting

This *is* actually enabled by default if you're offline. You generally want to use online authentication however, since games usually use your account data in some way (and won't work without it).

> Alternatively when there is no internet connection games downloaded should still be able to run. Why is an internet connection needed?

This is up to the specific game you're playing, not Heroic. Some work fine offline, some don't.

> If checking for updates is selected and you are offline and starting a game, it should not check for updates.

Update checks are done at Heroic startup. If they fail due to connection issues, Heroic will not retry them.

You can also already choose to launch without the update check (useful in a scenario where you know that the game needs an update, but don't have a connection to actually download that update). Heroic will automatically prompt you for this if you try to start a game with a pending update.

> Do not crash game when connection goes from online to offline during gameplay

A game's behavior to your connection dropping is not controlled (or controllable) by Heroic.

---

In conclusion, this is already implemented as well as we can do it.

**Comment by CommandMC** [comment]
> Is there a way to know whether the game needs the online connection?

Not *really*. Games can indicate that they support offline mode in their metadata, but this may be (and often is) incorrect

> it wasn't clear to me what the data connection actually does or what data is being shared.

The game gets a temporary token to use your Epic Games account. It can then use this for various purposes (most commonly used to simply log you into some other service with your Epic account)
If you turn on offline mode or are offline, the game won't get this token anymore. For games that don't need this token, there's no difference in functionality (they may even do other online-only processes like normal, assuming they don't require this token)
